src/Scriban/Functions/ArrayFunctions.cs

```diff
@@ -206,6 +198,41 @@ private static IEnumerable EachInternal(TemplateContext context, SourceSpan span
             }
         }
 
+        /// <summary>
+        /// Filters the input list according the supplied filter function.
+        /// </summary>
+        /// <param name="context">The template context</param>
+        /// <param name="span">The source span</param>
+        /// <param name="list">An input list</param>
+        /// <param name="function">The function used to test each elemement of the list</param>
+        /// <returns>Returns a new list which contains only those elements which match the filter function.</returns>
+        /// <remarks>
+        /// ```scriban-html
+        /// {{func large(x) ; ret x>=100 ; end}}
+        /// {{ [1, 200, 3,400] | array.each @large }}
+        /// ```
+        /// ```html
+        /// [200, 400]
+        /// ```
+        /// </remarks>
+        public static ScriptRange Filter(TemplateContext context, SourceSpan span, IEnumerable list, object function)
+        {
+            return ApplyFunction(context, span, list, function, FilterInternal);
+        }
+        private static IEnumerable FilterInternal(TemplateContext context, SourceSpan span, IEnumerable list, IScriptCustomFunction function, Type destType)
+        {
+            var arg = new ScriptArray(1);
+            foreach (var item in list)
+            {
+                var itemToTransform = context.ToObject(span, item, destType);
+                arg[0] = itemToTransform;
+                var itemTransformed = ScriptFunctionCall.Call(context, context.CurrentNode, function, arg);
+                if (itemTransformed is Boolean b && b)
```

Comment:
You should issue a context.ToBool on the itemTransformed to make sure that the type you receive is a boolean.

Reply:
got it - yes I can see that's  a much better approach